### ISSUE TITLE
[BUG] Get an error when trying to export SiteTemplate to file .pnp

### ISSUE THREAD

**Issue by maciejkukulka** [pnp/powershell#40]
### Notice
Want to copy one SPO site to another.

### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
Previous cmdlet: Get-PnPProvisioningTemplate works correctly. It exported whole site to .pnp file and restore it on another site. Now I have problem with Get-PnPSiteTemplate.

### Actual behavior 
```
PS C:\Users\maciej.kukulka> Get-PnPSiteTemplate -Out prototype.pnp -Force
WARNING: You need to be a SharePoint admin to extract Hub site Url.
Get-PnPSiteTemplate : The property or field 'ServerRelativePath' has not been initialized. It has not been requested or
 the request has not been executed. It may need to be explicitly requested.
At line:1 char:1
+ Get-PnPSiteTemplate -Out prototype.pnp -Force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Get-PnPSiteTemplate], PropertyOrFieldNotInitializedException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate
```

### Steps to reproduce behavior
Please include complete code samples in-line or linked from [gists](https://gist.github.com/)

### What is the version of the Cmdlet module you are running?
Oh, this is another problem I have ;)

```
PS C:\Users\maciej.kukulka> Install-Module SharePointPnPPowerShellOnline

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): a
PackageManagement\Install-Package : The following commands are already available on this system:'Invoke-PnPSearchQuery,
Add-PnPAlert,Add-PnPApp,Add-PnPApplicationCustomizer,Add-PnPClientSidePage,Add-PnPClientSidePageSection,Add-PnPClientSi
deText,Add-PnPClientSideWebPart,Add-PnPContentType,Add-PnPContentTypeToDocumentSet,Add-PnPContentTypeToList,Add-PnPCust
omAction,Add-PnPDocumentSet,Add-PnPEventReceiver,Add-PnPField,Add-PnPFieldFromXml,Add-PnPFieldToContentType,Add-PnPFile
,Add-PnPFolder,Add-PnPHtmlPublishingPageLayout,Add-PnPHubSiteAssociation,Add-PnPIndexedProperty,Add-PnPJavaScriptBlock,
Add-PnPJavaScriptLink,Add-PnPListItem,Add-PnPMasterPage,Add-PnPMicrosoft365GroupMember,Add-PnPMicrosoft365GroupOwner,Ad
d-PnPMicrosoft365GroupToSite,Add-PnPNavigationNode,Add-PnPOrgAssetsLibrary,Add-PnPOrgNewsSite,Add-PnPPublishingImageRen
dition,Add-PnPPublishingPage,Add-PnPPublishingPageLayout,Add-PnPRoleDefinition,Add-PnPSiteClassification,Add-PnPSiteCol
lectionAdmin,Add-PnPSiteCollectionAppCatalog,Add-PnPSiteDesign,Add-PnPSiteDesignTask,Add-PnPSiteScript,Add-PnPStoredCre
dential,Add-PnPTaxonomyField,Add-PnPTeamsChannel,Add-PnPTeamsTab,Add-PnPTeamsTeam,Add-PnPTeamsUser,Add-PnPTenantCdnOrig
in,Add-PnPTenantSequence,Add-PnPTenantSequenceSite,Add-PnPTenantSequenceSubSite,Add-PnPTenantTheme,Add-PnPUserToGroup,A
dd-PnPView,Add-PnPWebhookSubscription,Add-PnPWebPartToWebPartPage,Add-PnPWebPartToWikiPage,Add-PnPWikiPage,Approve-PnPT
enantServicePrincipalPermissionRequest,Clear-PnPDefaultColumnValues,Clear-PnPListItemAsRecord,Clear-PnPMicrosoft365Grou
pMember,Clear-PnPMicrosoft365GroupOwner,Clear-PnPRecycleBinItem,Clear-PnPTenantAppCatalogUrl,Clear-PnPTenantRecycleBinI
tem,Connect-PnPOnline,ConvertTo-PnPClientSidePage,Copy-PnPFile,Copy-PnPItemProxy,Deny-PnPTenantServicePrincipalPermissi
onRequest,Disable-PnPFeature,Disable-PnPSharingForNonOwnersOfSite,Disable-PnPSiteClassification,Disable-PnPTenantServic
ePrincipal,Disconnect-PnPOnline,Enable-PnPCommSite,Enable-PnPFeature,Enable-PnPSiteClassification,Enable-PnPTenantServi
cePrincipal,Export-PnPClientSidePage,Export-PnPClientSidePageMapping,Export-PnPTaxonomy,Export-PnPTermGroupToXml,Find-P
nPFile,Get-PnPAADUser,Get-PnPAccessToken,Get-PnPAlert,Get-PnPApp,Get-PnPAppAuthAccessToken,Get-PnPApplicationCustomizer
,Get-PnPAuditing,Get-PnPAuthenticationRealm,Get-PnPAvailableClientSideComponents,Get-PnPAvailableLanguage,Get-PnPAzureC
ertificate,Get-PnPClientSideComponent,Get-PnPClientSidePage,Get-PnPConnection,Get-PnPContentType,Get-PnPContentTypePubl
ishingHubUrl,Get-PnPContext,Get-PnPCustomAction,Get-PnPDefaultColumnValues,Get-PnPDeletedMicrosoft365Group,Get-PnPDocum
entSetTemplate,Get-PnPEventReceiver,Get-PnPException,Get-PnPFeature,Get-PnPField,Get-PnPFile,Get-PnPFileVersion,Get-PnP
Folder,Get-PnPFolderItem,Get-PnPFooter,Get-PnPGraphAccessToken,Get-PnPGraphSubscription,Get-PnPGroup,Get-PnPGroupMember
s,Get-PnPGroupPermissions,Get-PnPHideDefaultThemes,Get-PnPHomePage,Get-PnPHomeSite,Get-PnPHubSite,Get-PnPHubSiteChild,G
et-PnPIndexedPropertyKeys,Get-PnPInPlaceRecordsManagement,Get-PnPIsSiteAliasAvailable,Get-PnPJavaScriptLink,Get-PnPKnow
ledgeHubSite,Get-PnPLabel,Get-PnPList,Get-PnPListInformationRightsManagement,Get-PnPListItem,Get-PnPListRecordDeclarati
on,Get-PnPMasterPage,Get-PnPMicrosoft365Group,Get-PnPMicrosoft365GroupMembers,Get-PnPMicrosoft365GroupOwners,Get-PnPNav
igationNode,Get-PnPOffice365CurrentServiceStatus,Get-PnPOffice365HistoricalServiceStatus,Get-PnPOffice365ServiceMessage
,Get-PnPOffice365Services,Get-PnPOrgAssetsLibrary,Get-PnPOrgNewsSite,Get-PnPPowerShellTelemetryEnabled,Get-PnPProperty,
Get-PnPPropertyBag,Get-PnPPublishingImageRendition,Get-PnPRecycleBinItem,Get-PnPRequestAccessEmails,Get-PnPRoleDefiniti
on,Get-PnPSearchConfiguration,Get-PnPSearchCrawlLog,Get-PnPSearchSettings,Get-PnPSharingForNonOwnersOfSite,Get-PnPSite,
Get-PnPSiteClassification,Get-PnPSiteClosure,Get-PnPSiteCollectionAdmin,Get-PnPSiteCollectionTermStore,Get-PnPSiteDesig
n,Get-PnPSiteDesignRights,Get-PnPSiteDesignRun,Get-PnPSiteDesignRunStatus,Get-PnPSiteDesignTask,Get-PnPSitePolicy,Get-P
nPSiteScript,Get-PnPSiteScriptFromList,Get-PnPSiteScriptFromWeb,Get-PnPSiteSearchQueryResults,Get-PnPStorageEntity,Get-
PnPStoredCredential,Get-PnPSubWebs,Get-PnPTaxonomyItem,Get-PnPTaxonomySession,Get-PnPTeamsApp,Get-PnPTeamsChannel,Get-P
nPTeamsChannelMessage,Get-PnPTeamsTab,Get-PnPTeamsTeam,Get-PnPTeamsUser,Get-PnPTenant,Get-PnPTenantAppCatalogUrl,Get-Pn
PTenantCdnEnabled,Get-PnPTenantCdnOrigin,Get-PnPTenantCdnPolicies,Get-PnPTenantId,Get-PnPTenantRecycleBinItem,Get-PnPTe
nantSequence,Get-PnPTenantSequenceSite,Get-PnPTenantServicePrincipal,Get-PnPTenantServicePrincipalPermissionGrants,Get-
PnPTenantServicePrincipalPermissionRequests,Get-PnPTenantSite,Get-PnPTenantSyncClientRestriction,Get-PnPTenantTemplate,
Get-PnPTenantTheme,Get-PnPTerm,Get-PnPTermGroup,Get-PnPTermSet,Get-PnPTheme,Get-PnPTimeZoneId,Get-PnPUnifiedAuditLog,Ge
t-PnPUPABulkImportStatus,Get-PnPUser,Get-PnPUserOneDriveQuota,Get-PnPUserProfileProperty,Get-PnPView,Get-PnPWeb,Get-PnP
WebhookSubscriptions,Get-PnPWebPart,Get-PnPWebPartProperty,Get-PnPWebPartXml,Get-PnPWebTemplates,Get-PnPWikiPageContent
,Grant-PnPHubSiteRights,Grant-PnPSiteDesignRights,Grant-PnPTenantServicePrincipalPermission,Import-PnPTaxonomy,Import-P
nPTermGroupFromXml,Import-PnPTermSet,Initialize-PnPPowerShellAuthentication,Install-PnPApp,Invoke-PnPQuery,Invoke-PnPSi
teDesign,Invoke-PnPSPRestMethod,Invoke-PnPWebAction,Measure-PnPList,Measure-PnPWeb,Move-PnPClientSideComponent,Move-PnP
File,Move-PnPFolder,Move-PnPItemProxy,Move-PnPListItemToRecycleBin,Move-PnPRecycleBinItem,New-PnPAzureCertificate,New-P
nPExtensibilityHandlerObject,New-PnPGraphSubscription,New-PnPGroup,New-PnPList,New-PnPMicrosoft365Group,New-PnPPersonal
Site,New-PnPSite,New-PnPTeamsApp,New-PnPTeamsTeam,New-PnPTenantSequence,New-PnPTenantSequenceCommunicationSite,New-PnPT
enantSequenceTeamNoGroupSite,New-PnPTenantSequenceTeamNoGroupSubSite,New-PnPTenantSequenceTeamSite,New-PnPTenantSite,Ne
w-PnPTenantTemplate,New-PnPTerm,New-PnPTermGroup,New-PnPTermLabel,New-PnPTermSet,New-PnPUPABulkImportJob,New-PnPUser,Ne
w-PnPWeb,Publish-PnPApp,Read-PnPTenantTemplate,Register-PnPAppCatalogSite,Register-PnPHubSite,Register-PnPManagementShe
llAccess,Remove-PnPAlert,Remove-PnPApp,Remove-PnPApplicationCustomizer,Remove-PnPClientSideComponent,Remove-PnPClientSi
dePage,Remove-PnPContentType,Remove-PnPContentTypeFromDocumentSet,Remove-PnPContentTypeFromList,Remove-PnPCustomAction,
Remove-PnPDeletedMicrosoft365Group,Remove-PnPEventReceiver,Remove-PnPField,Remove-PnPFieldFromContentType,Remove-PnPFil
e,Remove-PnPFileVersion,Remove-PnPFolder,Remove-PnPGraphSubscription,Remove-PnPGroup,Remove-PnPHomeSite,Remove-PnPHubSi
teAssociation,Remove-PnPIndexedProperty,Remove-PnPJavaScriptLink,Remove-PnPKnowledgeHubSite,Remove-PnPList,Remove-PnPLi
stItem,Remove-PnPMicrosoft365Group,Remove-PnPMicrosoft365GroupMember,Remove-PnPMicrosoft365GroupOwner,Remove-PnPNavigat
ionNode,Remove-PnPOrgAssetsLibrary,Remove-PnPOrgNewsSite,Remove-PnPPropertyBagValue,Remove-PnPPublishingImageRendition,
Remove-PnPRoleDefinition,Remove-PnPSearchConfiguration,Remove-PnPSiteClassification,Remove-PnPSiteCollectionAdmin,Remov
e-PnPSiteCollectionAppCatalog,Remove-PnPSiteDesign,Remove-PnPSiteDesignTask,Remove-PnPSiteScript,Remove-PnPStorageEntit
y,Remove-PnPStoredCredential,Remove-PnPTaxonomyItem,Remove-PnPTeamsApp,Remove-PnPTeamsChannel,Remove-PnPTeamsTab,Remove
-PnPTeamsTeam,Remove-PnPTeamsUser,Remove-PnPTenantCdnOrigin,Remove-PnPTenantSite,Remove-PnPTenantTheme,Remove-PnPTermGr
oup,Remove-PnPUser,Remove-PnPUserFromGroup,Remove-PnPView,Remove-PnPWeb,Remove-PnPWebhookSubscription,Remove-PnPWebPart
,Remove-PnPWikiPage,Rename-PnPFile,Rename-PnPFolder,Request-PnPAccessToken,Request-PnPReIndexList,Request-PnPReIndexWeb
PageConversionLog,Save-PnPTenantTemplate,Send-PnPMail,Set-PnPApplicationCustomizer,Set-PnPAppSideLoading,Set-PnPAuditin
g,Set-PnPAvailablePageLayouts,Set-PnPClientSidePage,Set-PnPClientSideText,Set-PnPClientSideWebPart,Set-PnPContext,Set-P
nPDefaultColumnValues,Set-PnPDefaultContentTypeToList,Set-PnPDefaultPageLayout,Set-PnPDocumentSetField,Set-PnPField,Set
-PnPFileCheckedIn,Set-PnPFileCheckedOut,Set-PnPFolderPermission,Set-PnPFooter,Set-PnPGraphSubscription,Set-PnPGroup,Set
-PnPGroupPermissions,Set-PnPHideDefaultThemes,Set-PnPHomePage,Set-PnPHomeSite,Set-PnPHubSite,Set-PnPIndexedProperties,S
et-PnPInPlaceRecordsManagement,Set-PnPKnowledgeHubSite,Set-PnPLabel,Set-PnPList,Set-PnPListInformationRightsManagement,
Set-PnPListItem,Set-PnPListItemAsRecord,Set-PnPListItemPermission,Set-PnPListPermission,Set-PnPListRecordDeclaration,Se
t-PnPMasterPage,Set-PnPMicrosoft365Group,Set-PnPMinimalDownloadStrategy,Set-PnPPropertyBagValue,Set-PnPRequestAccessEma
ils,Set-PnPSearchConfiguration,Set-PnPSearchSettings,Set-PnPSite,Set-PnPSiteClosure,Set-PnPSiteDesign,Set-PnPSitePolicy
,Set-PnPSiteScript,Set-PnPStorageEntity,Set-PnPTaxonomyFieldValue,Set-PnPTeamifyPromptHidden,Set-PnPTeamsChannel,Set-Pn
PTeamsTab,Set-PnPTeamsTeam,Set-PnPTeamsTeamArchivedState,Set-PnPTeamsTeamPicture,Set-PnPTenant,Set-PnPTenantAppCatalogU
rl,Set-PnPTenantCdnEnabled,Set-PnPTenantCdnPolicy,Set-PnPTenantSite,Set-PnPTenantSyncClientRestriction,Set-PnPTermGroup
,Set-PnPTermSet,Set-PnPTheme,Set-PnPTraceLog,Set-PnPUserOneDriveQuota,Set-PnPUserProfileProperty,Set-PnPView,Set-PnPWeb
,Set-PnPWebhookSubscription,Set-PnPWebPartProperty,Set-PnPWebPermission,Set-PnPWebTheme,Set-PnPWikiPageContent,Submit-P
nPSearchQuery,Submit-PnPTeamsChannelMessage,Sync-PnPAppToTeams,Test-PnPListItemIsRecord,Test-PnPTenantTemplate,Uninstal
l-PnPApp,Unpublish-PnPApp,Unregister-PnPHubSite,Update-PnPApp,Update-PnPSiteClassification,Update-PnPTeamsApp'. This mo
dule 'SharePointPnPPowerShellOnline' may override the existing commands. If you still want to install this module 'Shar
ePointPnPPowerShellOnline', use -AllowClobber parameter.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\2.2.5\PSModule.psm1:9711 char:34
+ ... talledPackages = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package],
   Exception
    + FullyQualifiedErrorId : CommandAlreadyAvailable,Validate-ModuleCommandAlreadyAvailable,Microsoft.PowerShell.Pack
   ageManagement.Cmdlets.InstallPackage
```

**Comment by maciejkukulka** [comment]
My PnPTraceLog at the end looks like:


```
w Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext()
--
ServerErrorCode: -2147024809
ServerErrorTypeName: System.ArgumentException
ServerErrorTraceCorrelationId: 8f62939f-2016-b000-9521-0f57273bd84f
ServerErrorValue:
ServerErrorDetails:
.	0ms
powershell.exe Warning: 0 : 2020-12-02 16:04:52.4575	[Site Footer]	[14]	[Information]	File is not necessarily located in the current web. Not retrieving /sites/MDMS	1029ms	7c24967c-2373-44e8-9c90-4987efa83241
powershell.exe Error: 0 : 2020-12-02 16:04:56.1187	[PnP.Framework]	[0]	[Error]	ExecuteQuery threw following exception: System.Net.WebException: Serwer zdalny zwrócił błąd: (401) Bez autoryzacji.
w System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
w System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
w System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
w Microsoft.SharePoint.Client.SPWebRequestExecutor.<ExecuteAsync>d__20.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
w System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
w Microsoft.SharePoint.Client.ClientContext.<GetFormDigestInfoPrivateAsync>d__37.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
w System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
w Microsoft.SharePoint.Client.ClientContext.<EnsureFormDigestAsync>d__36.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
w System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
w Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__28.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
w System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
w System.Runtime.CompilerServices.TaskAwaiter.GetResult()
w Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext()
ServerErrorTraceCorrelationId: 8f62939f-30f3-b000-4e2f-fa5a55c9d0ba
.	0ms


```

**Comment by erwinvanhunen** [comment]
Regarding the install issue: you're trying to install the legacy version if you use SharePointPnPPowerShellOnline. Most of the cmdlets have in the legacy (e.g. SharePointPnPPowerShellOnline) and the new version (e.g. PnP.PowerShell) have the same name, which causes conflicts if you install them. E.g. you should either the legacy -or- the newer version installed.

Regarding the template creation you're seeing when using the new version: what way of connecting do you use? It sounds like a rights problem (authentication has changed -considerably- in PnP.PowerShell).

**Comment by maciejkukulka** [comment]
Hi @erwinvanhunen 
I deleted all modules and installed new version of PS (7.1.0). Still get an error from first post.
I am Global Admin of Tenant and Admin of Hub Site.

![2020-12-07 11_56_24-Administrator_ PowerShell](https://user-images.githubusercontent.com/67901399/101342856-6fa6b680-3883-11eb-8a94-bbd48320405c.png)

**Comment by erwinvanhunen** [comment]
unfortunately I'm not able to reproduce this on my environment.

A few questions.

Can you try:

```powershell
Get-PnPSiteTemplate -Handlers WebSettings
```

That helps us to determine if the issue occurs when extracting websettings (most likely it fails there).

I assume this site is associated with a hubsite?

Do you see the site if you enter:

```powershell
Get-PnPHubSiteChild -Identity https://url.of.your.hubsite
```
?

**Comment by maciejkukulka** [comment]
I tried both cmdlets and here is the result:
![image](https://user-images.githubusercontent.com/67901399/101498374-50d01f00-396c-11eb-9921-dc7e8b4b36db.png)

**Comment by erwinvanhunen** [comment]
As the cmdlet mentions, you cannot use a device login authentication and use 'admin' level cmdlets that work on tenant level without actually connecting to the tenant admin site. I see in your screen shot that you actually connect to your tenant admin site, but immediately after that you connect back to your hubsite. That will basically reset the connection back to the your hubsite and PnP PowerShell has no understanding of your tenant site. This has to do with the way Microsoft implements authentication and how we can acquire an access token behind the scenes to connect to the sites. If you require in your script to mix and match the cmdlets, -and- use a device login- then  you can do this:

```powershell
$connection = Connect-PnPOnline -Url "https://contoso.sharepoint.com" -PnPManagementShell -ReturnConnection
$adminConnection = Connect-PnPOnline -Url "https://contoso-admin.sharepoint.com" -PnPManagement -ReturnConnection

Get-PnPSite -Connection $connection
Get-PnPHubSiteChild -Identity "https://contoso.sharepoint.com/sites/Products" -Connection $adminConnection
```
*Notice that the second `Connect-PnPOnline` line connects to your tenant admin site.*

What you can do, alternatively as you have the the PnPManagementShell multi-tenant application already registered in your Azure AD:

```powershell
Connect-PnPOnline -Url "https://contoso.sharepoint.com/your/hubsite" -Credentials (Get-Credential)
Get-PnPSite 
Get-PnPHubSiteChild -Identity "https://contoso.sharepoint.com/your/hubsite"
```

Using the first example you build basically two connections, one to your normal site, the other to your tenant admin site and you tell the cmdlets which site to use.
Using the second example we still use, behind the scenes, the PnP Management Shell azure AD application to connect to your tenant. However, because you provide credentials, we can acquire an access token behind the scenes towards the Tenant Administration site, and the cmdlets will be able to switch connections behind the scenes automatically. The second example however is not working if your tenant uses multi-factor authentication.

If MFA is enabled then another alternative is to use -WebLogin. This will fall back to legacy authentication, which still needs to be enabled on your tenant (most tenants have this still enabled currently). If you plan to use -WebLogin we recommend you to upgrade first to the latest build as we make some improvements yesterday and today.

```powershell
Connect-PnPOnline -Url "https://contoso.sharepoint.com/your/hubsite" -UseWebLogin
Get-PnPSite 
Get-PnPHubSiteChild -Identity "https://contoso.sharepoint.com/your/hubsite"
```

**Comment by maciejkukulka** [comment]
Hi @erwinvanhunen 
first of all I want to thenk you for your help. Really appreciate it.

Important thing. I want to get Site Template from site called: MDMS. Not hub site: Products.

When I used Get-PnpProvisioningTemplate about a month ago it works totally fine. There wasn't any issue with HubSite.

I updated module PnP.PowerShell and I tried all 3 solutions you written above. Sadly none of them works.

Solution no. 1 
![ver-1](https://user-images.githubusercontent.com/67901399/101915845-75273800-3bc6-11eb-964a-d5166c4dbd9d.png)

Solution no. 2
![ver-2](https://user-images.githubusercontent.com/67901399/101915903-8708db00-3bc6-11eb-83c5-1dbc3d79fb81.png)

Solution no. 3
![ver-3](https://user-images.githubusercontent.com/67901399/101915930-8f611600-3bc6-11eb-9d5f-315747403189.png)

Maybe I need to turn off MFA to make these operations?

**Comment by erwinvanhunen** [comment]
Regarding Solution 1: we are aware of the issue you're seeing there where it is not being able to write to the output. That seems to be an issue in both PowerShell ISE and VSCode. Having said that, please give it a try with the nightly build that will be available tomorrow (0.3.8-nightly) as we've made some changes to it. 
However, just for your information: the moment you have a successful connection to your tenant (e.g. the first connection method did work) you can use `Get-PnPHubSiteChild`. The parameter Identity points to the site for which you want to retireve the hubsites, regardless if you have access to the actual site or not (this is because we ask this information from the tenant administration site behind the scenes and not the actual site).

Regarding Solution 2: okay, nothing we can do about that. If your admin requires MFA then credential auth will not work.

Regarding Solution 3: It almost looks like you do not have access to the site https://yourtenant.sharepoint.com/sites/products. Notice that being a tenant admin does not automatically grant you access to a site.  That would also explain the issues you have in solution 1. Through the Microsoft 365 Administration portal you can grant yourself access to the site. Could you double check there if you have access?
